### DISCUSSION TITLE
Subdirectories should be ignored during beatmap import

### DISCUSSION THREAD

**Discussion by Joppe27** [ppy/osu#18592] [Q&A] [answered]
When importing beatmaps to lazer, having a subdirectory in the stable "Songs" folder causes some issues:

```
osu!/
├─ Songs/
   ├─ beatmap1/
   ├─ beatmap2/
   ├─ beatmap3/
   ├─ Moar Songs/
       ├─ beatmap4/
       ├─ beatmap5/
       ├─ beatmap6/
```
I you were to import this example, all of the .osu files in the beatmap4/5/6 folders would get grouped up and assigned to a random (?) song in that subfolder. This can create *enormous* beatmaps with huge amounts of .osu's (or diffs, whatever you want to call it) like in this example: https://joppe27.s-ul.eu/jrlvQ5IQ.

I know this is user error and that the subdirectory shouldn't be there to begin with, but it would be nice if lazer could behave the same as stable here (as in ignoring the subfolder during import) or at least give a warning. That way the lazer database won't get full of dupe .osu's slowing down song select, plus users also won't have to wait for all of them to import twice.

**Comment by peppy** [comment, marked as answer]
Tracking at https://github.com/ppy/osu/issues/18615
